web/controllers/api/v1/message_controller.ex

```diff
@@ -116,16 +115,72 @@ defmodule Kastlex.API.V1.MessageController do
             true = size > max_bytes # assert
             ## fetched back an incomplete message
             ## try again with the exact message size
-            do_fetch(sock_pid, req_fun, size, offset)
-          messages ->
-            resp = %{errorCode: errorCode,
-                     highWmOffset: highWmOffset,
+            do_fetch(sock_pid, req_fun, size, offset, type)
+          messages when type == "json" ->
+            resp = %{highWmOffset: highWmOffset,
                      size: size,
                      messages: messages_to_map(messages)
                     }
             {:ok, resp}
+          messages when type == "binary" ->
+            {:kafka_message, _offset, _magic_byte, _attributes, _key, value, _crc} = hd(messages)
```

Comment:
can we make use of the record ?
soon we'll add another field `timestamp` to the message, then this will crash